src/parser/parser.py

```diff
@@ -0,0 +1,60 @@
+import slate
+
+
+import logging
+
+# import numpy
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Article:
+    """Class for storing articles.
+    """
+
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+def count_articles(pdf_text):
+    """Identifies articles and returns a list of Article objects.
+
+    Args:
+        pdf_text (string): contains the PDF text where articles 
+        are to be identified.
+
+    Returns:
+        list: article objects
+    """
+    articles = []
+    article_text = ""
+    article_count = 1
+    i = 0
+    while i < len(pdf_text) - 1:
```

Comment:
For enumerate might make this loop much cleaner. Enumerating gives both the index you are currently in and the value. https://treyhunner.com/2016/04/how-to-loop-with-indexes-in-python/#enumerate

Reply:
While it is much cleaner, the behavior is not the same. I can't add to the variable `i` from inside the loop when using a for, thus I use a `while` loop.

Reply:
You're right. Though, sometimes a continue statement can serve a similar purpose

Reply:
Yes but a continue does not work in this case. I'm adding a one to `i` in order to skip adding the article number to the text.